examples/tool_chat_template_mistral3.jinja

```diff
@@ -29,8 +29,16 @@
 
 {%- set user_messages = loop_messages | selectattr("role", "equalto", "user") | list %}
 
-{%- for message in loop_messages | rejectattr("role", "equalto", "tool") | rejectattr("role", "equalto", "tool_results") | selectattr("tool_calls", "undefined") %}
-    {%- if (message["role"] == "user") != (loop.index0 % 2 == 0) %}
+{%- set filtered_messages = [] %}
+{%- for message in loop_messages %}
+    {%- if message["role"] not in ["tool", "tool_results"] and not message.get("tool_calls") %}
```

Comment:
It doesn't seem fundamentally different from `rejectattr("role", "equalto", "tool") | rejectattr("role", "equalto", "tool_results") | selectattr("tool_calls", "undefined")`.